### PR TITLE
Update FatFs and add USB-MSC firmware update example

### PR DESCRIPTION
**Pull request by salkinium**
This converts FatFs by Chan to our submodule mechanism and updates to the latest version with all patches applied.

The example combines TinyUSB MSC with a >64kB FAT12 ramdisk to create a simple firmware update mechanism that should work without any special tooling (hopefully on any operating system).

In addition there are some minor fixes and updates.

@rleh Could you test this on linux?

### CODE REVIEW

File: ext/chan/ffconf.h.in [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+#define FF_USE_FORWARD	1
+/* This option switches f_forward() function. (0:Disable or 1:Enable) */
+
+
+/*---------------------------------------------------------------------------/
+/ Locale and Namespace Configurations
+/---------------------------------------------------------------------------*/
+
+#define FF_CODE_PAGE	932
```
Comment (rleh) [inline ext/chan/ffconf.h.in]: `932 - Japanese (DBCS)`
Sure?

File: ext/chan/module.lb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    env.collect(":build:path.include", "modm/ext")
+    env.outbasepath = "modm/ext/fatfs"
+
+    env.copy("fatfs/full/source/diskio.h", "diskio.h")
+    env.copy("fatfs/full/source/ff.h", "ff.h")
+    env.copy("fatfs/full/source/ff.c", "ff.c")
+    env.copy("fatfs/full/source/ffunicode.c", "ffunicode.c")
+    env.copy("fatfs/full/LICENSE.txt", "LICENSE.txt")
+
+    env.template("ffconf.h.in")
```
Comment (rleh) [inline ext/chan/module.lb]:
```suggestion
    env.copy("ffconf.h")
```
Not template functionality is used...?

File: ext/hathach/module.lb [hunk 3]
```diff
@@ -158,8 +158,6 @@ def build(env):
     if target.platform == "stm32":
         irq_data = env.query(":platform:usb:irqs")
         irqs = irq_data["port_irqs"][speed]
-        if irq_data["is_remap"]:
-            tusb_config["STM32F3_IRQ_REMAP"] = 1
```
Comment (rleh) [inline ext/hathach/module.lb:162]: Fixed upstream?

File: ext/chan/ffconf.h.in (hunk 1 above)
Comment (rleh) [reply ext/chan/ffconf.h.in]: Default?

File: ext/hathach/module.lb (hunk 3 above)
Comment (salkinium) [reply ext/hathach/module.lb:162]: Yes.

File: ext/chan/ffconf.h.in (hunk 1 above)
Comment (salkinium) [reply ext/chan/ffconf.h.in]: It's the default and the code pages are not enabled, so this does not matter.

File: ext/chan/module.lb (hunk 2 above)
Comment (salkinium) [reply ext/chan/module.lb]: Fixed.

**Comment by rleh** [comment]
> @rleh Could you test this on linux?

Tomorrow.

**Review by rleh — APPROVED** [review]
Works on my machine.

Nice work, thank you @salkinium!

**Pull request merged**